src/ObjectUtils/SymbolsFile.cpp

```diff
@@ -52,4 +54,29 @@ ErrorMessageOr<std::unique_ptr<SymbolsFile>> CreateSymbolsFile(
   return ErrorMessage{error_message};
 }
 
+[[nodiscard]] std::vector<std::filesystem::path> GetAllowedSymbolFilenamesForModule(
+    const std::filesystem::path& module_path,
+    const orbit_grpc_protos::ModuleInfo::ObjectFileType& object_file_type) {
+  std::string sym_ext;
+  switch (object_file_type) {
+    case orbit_grpc_protos::ModuleInfo::kElfFile:
+      sym_ext = ".debug";
+      break;
+    case orbit_grpc_protos::ModuleInfo::kCoffFile:
+      sym_ext = ".pdb";
+      break;
+    default:
+      ORBIT_ERROR("Unknown object file type");
+      return {module_path.filename()};
```

Comment:
Is this needed? Wouldn't the compiler complain if you had unlisted enum values and no default case?

Reply:
Yes, the compiler complains. I designed it with default, because: 
This comes from module.proto
```
enum ObjectFileType {
    kUnknown = 0;
    kElfFile = 1;
    kCoffFile = 2;
}
ObjectFileType object_file_type = 10;
```
So there is the kUnknown option (and then since its from proto, there is also generated: `ObjectFileType_MIN` & `ObjectFileType_MAX`). 

Two questions: 
1. Are you suggesting to not use default, to get a compile error the next time a value is added to the enum? Because the longer I think about it, the more I prefer the version without default. (I would do all cases and explicit fallthroughs). 
2. It should theoretically never be `kUnknown`, but maybe it is, maybe when a user opens an old capture file where the serialized module proto does not contain the `object_file_type` field -> then it would default to kUnknown. I am not sure if this actually can happen. So crash (CHECK/UNREACHABLE) if kUnknown, or not?

Reply:
Regarding 1: Yes, the idea would be to get a compile error such that when anyone adds a value to `ObjectFileType` the compiler will notify them of this code.

Regarding 2: I would return early for kUnknown (no crash, check, unreachable) - similar to what's currently the default case.

Reply:
Oh, I just saw your change. I didn't know about these sentinel values that protobuf defines. Not sure what's best practice in this case. I will do some research.

Reply:
According to https://groups.google.com/g/protobuf/c/YefxUmgxvg4 having a default case is recommended. Sorry for derailing this... :see_no_evil:

Reply:
I went with the version without default and ERROR (recoverable) instead of UNREACHABLE in the case of kUnknown